src/Integration/Helpers/SonarLintOutputLogger.cs

```diff
@@ -29,15 +29,12 @@ namespace SonarLint.VisualStudio.Integration
     public class SonarLintOutputLogger : ILogger
     {
         private readonly IServiceProvider serviceProvider;
+        private bool shouldLogDebug = false;
 
         [ImportingConstructor]
         public SonarLintOutputLogger([Import(typeof(SVsServiceProvider))]IServiceProvider serviceProvider)
         {
-            if (serviceProvider == null)
-            {
-                throw new ArgumentNullException(nameof(serviceProvider));
-            }
-            this.serviceProvider = serviceProvider;
+            this.serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
```

Comment:
FYI MEF will never pass a null in this case (it can for `ImportMany`, but here the composition will fail). We have this pattern in quite a few places, but I don't think it helps.

Reply:
Yeah I removed it in future PRs 👍